### PR TITLE
.NET 5 upgrade; Debug mode

### LINKED ISSUE #41: DoD: Test/debug mode
To facilitate easier testing, I am suggesting a test/debug mode that would modify the following:
- Disable "building must be adjacent to another" rule
- Allow all building types to be placed rather than any random 2

This would help alleviate difficulty in reproducing manual test cases.

Tagging SM and QA for discussion/approval - @JSQJ @kenwongz

### PR DESCRIPTION
**Pull request by CaeNGnp**
### 1. Debug mode

Introduces a debug mode that disables certain built-in rules to make debugging easier.

Resolves #41 

### 2. Upgrading to .NET 5

Move from legacy .NET Framework 4.8 to .NET 5 (aka .NET Core 5).

Reasons:
- C# 9 features (https://devblogs.microsoft.com/dotnet/c-9-0-on-the-record/),
including nullable reference type and more
- Access to better code coverage tools
- Much more maintainable assembly information file (auto-generated)
- New and cleaner `.csproj` format
- Streamlined building tools (`dotnet` instead of `msbuild`)
- ⚠️ Cross-platform support ⚠️

Further reading:
- https://docs.microsoft.com/en-us/dotnet/core/whats-new/dotnet-5
- https://blog.ndepend.com/not-planning-now-to-migrate-yout-net48-legacy-is-certainly-a-mistake/

Tests:
- Manual; no obvious regressions spotted
- Unit tests pass

⚠️ **This is a big change that can potentially break stuff. Before testing, make sure:**

1. VS2019 >=v16.11 is installed (according to docs it's 16.8 minimum but I have yet to test it) 
2. To close VS IDE after pulling
3. To delete `.vs` folder to avoid any caching errors
 (or better - `git clean -fxd`)
4. Relaunch VS

### CODE REVIEW

**Comment by CaeNGnp** [comment]
CD is broken at the moment.

**Comment by CaeNGnp** [comment]
[Fixed](https://github.com/kenwongz/SimpCity/releases/tag/dev-28db66af7de9b7fd3ca61382b36207b249c6bc62)

File: SimpCityTests/packages.config [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="MSTest.TestAdapter" version="2.1.2" targetFramework="net472" />
-  <package id="MSTest.TestFramework" version="2.1.2" targetFramework="net472" />
+  <package id="MSTest.TestAdapter" version="2.2.8" targetFramework="net472" />
+  <package id="MSTest.TestFramework" version="2.2.8" targetFramework="net472" />
 </packages>
```
Comment (CaeNGnp) [inline SimpCityTests/packages.config]: File is no longer needed with new `csproj`

**Review by kenwongz — APPROVED** [review]
Thanks for the fix

**Review by JSQJ — APPROVED** [review]

**Pull request merged**